log/src/main/java/kcl/teamIndexZero/traffic/log/Outputs/Formatter_CSV.java

```diff
@@ -47,11 +44,13 @@ public String format(String origin_name, int log_level, Long log_number, Log_Tim
      */
     @Override
     public String format(String origin_name, Log_TimeStamp time_stamp, Long log_number, Exception e) {
-        StringWriter sw = new StringWriter();
-        PrintWriter pw = new PrintWriter(sw);
-        e.printStackTrace(pw);
-        String s = log_number.toString() + ";" + time_stamp.getDate() + ";" + time_stamp.getTime() + ";EXCEPTION;" + origin_name + "; " + sw.toString();
-        //TODO check exception details format
+        String s = String.format( "%d;%s;%s;EXCEPTION;%s;%s",
+                log_number,
+                time_stamp.getDate(),
+                time_stamp.getTime(),
+                origin_name,
+                e.toString()
```

Comment:
Well, that's more like a generic design flow to me - basically we are packing something into CSV which may be unsafe (i.e. break columns). After actually checking into the CSV spec, what i can say it uses double quotes to escape sequences, i.e. this is a part of [grammar for CSV as per spec ](http://tools.ietf.org/html/rfc4180)

```
   escaped = DQUOTE *(TEXTDATA / COMMA / CR / LF / 2DQUOTE) DQUOTE
```

For this very case I agree it is not too strict - but in arbitrary code dealing with some sort of export (lots of lifelike tasks) it may be a problem. 

I'd say we'd rather wrap fields in double quotes - but now we need to escape them too :))) which seems a bit easier, if I got format right - you just can have `2DQUOTE` for parser to identify it as an escaped quote. 

wdyt?